tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs

```diff
@@ -21,21 +22,28 @@ public InfiniteTracingTestsBase(TFixture fixture, ITestOutputHelper output):base
             _fixture.TestLogger = output;
 
             _fixture.AddCommand($"InfiniteTracingTester StartAgent");
+
+            _fixture.AddCommand("RootCommands DelaySeconds 5"); // give the agent time to warm up
+
             _fixture.AddCommand($"InfiniteTracingTester Make8TSpan");
-            _fixture.AddCommand($"InfiniteTracingTester Wait");
 
+            _fixture.AddCommand("RootCommands DelaySeconds 5"); // brief wait for the infinite trace response to come back
 
-            _fixture.Actions
-            (
+            _fixture.AddActions(
                 setupConfiguration: () =>
                 {
                     var configModifier = new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath);
 
                     configModifier.ForceTransactionTraces()
                     .EnableDistributedTrace()
-                    .EnableInfinteTracing(_fixture.TestConfiguration.TraceObserverUrl)
+                    .EnableInfiniteTracing(_fixture.TestConfiguration.TraceObserverUrl, _fixture.TestConfiguration.TraceObserverPort)
                     .SetLogLevel("finest");
+                },
+                exerciseApplication: () =>
+                {
+                    _fixture.AgentLog.WaitForLogLine(AgentLogBase.TransactionTransformCompletedLogLineRegex, TimeSpan.FromMinutes(2));
```

Comment:
This log line is recorded before Infinite Tracing has had a chance to send the transaction results, I would suggest waiting for a log line from our gRPC code that indicates spans have been sent

Reply:
Re-worked